Formula/tuir.rb

```diff
@@ -0,0 +1,23 @@
+class Tuir < Formula
+  include Language::Python::Virtualenv
+
+  desc "Command-line Reddit client"
+  homepage "https://gitlab.com/ajak/tuir"
+  url "https://gitlab.com/ajak/tuir/-/archive/master/tuir-master.tar.gz"
+  version "1.28.1"
+  sha256 "975d8cca39797eb604023d916370d2195fde363f397101a00f2927888be9eea5"
+
+  depends_on "python"
+
+  def install
+    venv = virtualenv_create(libexec, "python3")
+    system libexec/"bin/pip", "install", "-v", "--no-binary", ":all:",
+      "--ignore-installed", buildpath
+    system libexec/"bin/pip", "uninstall", "-y", name
+    venv.pip_install_and_link buildpath
+  end
+
+  test do
+    assert_equal "tuir 1.28.1\n", shell_output("#{bin}/tuir --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.